### PR TITLE
Extend pitch fly mode to swimming

### PR DESCRIPTION
**Pull request by random-geek**
This basically just renames everything from "pitch fly" to "pitch move", and makes it apply when in liquid.

I did make one fairly minor change to the pitch movement code, and that is that one can now travel at full speed forward when flying/swimming along the ground looking downward. This alleviates some frustration with swimming in pitch mode, but also has weird behavior at the edges of blocks due to how `touching_ground` seems to work.

### CODE REVIEW

File: src/client/localplayer.cpp [hunk 1]
```diff
@@ -727,10 +735,13 @@ v3f LocalPlayer::getEyeOffset() const
 
 // 3D acceleration
 void LocalPlayer::accelerate(const v3f &target_speed, const f32 max_increase_H,
-		const f32 max_increase_V, const bool use_pitch)
+		const f32 max_increase_V, bool use_pitch)
 {
 	const f32 yaw = getYaw();
 	const f32 pitch = getPitch();
+	if (use_pitch && touching_ground && pitch > 0.0f)
+		// Prevent player from slowing down when looking at the ground in pitch mode
```
Comment (SmallJoker) [inline src/client/localplayer.cpp]: Even though this is technically correct, please add brackets around this if-block.

**Comment by paramat** [comment]
I don't see this as needed or of much use, swimming is very different to flying and occurs in a much more restricted space (32 nodes in most mapgens). Very often when swimming you want to look in a different direction to movement direction, down at the seabed for example.
> // Don't sink when swimming in pitch mode

I disagree with this, makes sinking too easily avoidable :-1:

**Comment by Extex101** [comment]
Could you provide a video?

**Comment by paramat** [comment]
> one can now travel at full speed forward when flying/swimming along the ground looking downward. This alleviates some frustration with swimming in pitch mode

This doesn't seem necessary. When flying at a downward angle, before coming into contact with ground the horizontal component of velocity will be a certain value, then when contacting ground the horizontal component suddenly jumps to full speed.
This will be increasingly weird when pitch angle is near straight down, when you contact ground you won't want to suddenly move at full speed horizontally.
It's more intuitive for the horizontal component of velocity to remain unchanged, and is consistent with general player movement when walking against an obstruction at an angle.

What is the cause of the 'frustration' you mention?

**Comment by paramat** [comment]
So i'm :-1: due to that (my previous comment) and the disabling of sinking.

**Comment by random-geek** [comment]
@paramat and all others interested:
Just to clarify some things...

1. **Sinking in pitch mode still occurs when the player is mot moving at all.**
2. As you know, sinking in "normal" swimming mode is constant except when the player is pressing keys to ascend or descend. This prevents ascent being too slow or descent being too fast. Originally I had kept this behavior for pitch swimming; however, pressing "forward" while looking upwards was then much slower than ascending normally with the space bar. One possible solution is to calculate some sort of "sinking factor" based on the pitch of the speed vector, which is doable, but quite a strange mechanic.
3. Disabling sinking when swimming in pitch mode preserves the player's facing direction—if a player centers their cross-hairs on a certain location, they expect to move in that direction when they press the forward key.
4. The "frustration" I was talking about is with moving along the ground in pitch swimming mode. For example, when actively digging clay, the player (me) is often walking on the sea floor rather than swimming. In "normal" pitch mode, the player slows down significantly when looking at the ground, which can be quite infuriating considering the already slow pace when moving underwater. If this feature is unwanted, I am more than happy to remove it.

Hope that all makes sense.

**Comment by DS-Minetest** [comment]
>Very often when swimming you want to look in a different direction to movement direction, down at the seabed for example.

It's actually the same with flying, you want to look at the ground or at the ceiling.

Imo as long as you are sinking while not swimming, it's ok. In non pitch swim it's the same. I don't see a difference in pressing jump every second for half a second or whatever or swimming in some direction without sinking.

For consistency this PR makes sense. Other games that have pitch flight (eg. Sauerbraten) also have pitch swim, too.

**Comment by paramat** [comment]
> Sinking in pitch mode still occurs when the player is mot moving at all.

Sorry i misunderstood, that's fine then. Edited my post above.
> one can now travel at full speed forward when flying/swimming along the ground looking downward.

I'm still :-1: for this.

**Comment by benrob0329** [comment]
What other games do is not a non-argument. Being a game engine, MT needs to look towards already existing games and engines for known and intuitive behavior.

**Comment by paramat** [comment]
I meant that specifically for pitch fly and swim.
MT doesn't 'need' to, it's completely possible to create an excellent and user-friendly game without thinking about other games at all. What is 'intuitive' is, by definition (inner tuition), known from within users, not from other games. 'known' behaviour is not necessarily 'right' behaviour for MT, if it is that's fine but it should always be questioned.

**Comment by lhofhansl** [comment]
Personally I like this. It's optional, so why not add this. Swimming before felt unnatural (but that's an opinion).
Unless it is a detriment to some other feature or allows cheating in some way, I don't see why shouldn't be added.

**Comment by random-geek** [comment]
This is ready for review, I think.

**Comment by paramat** [comment]
Thanks, no objections, it's also fairly simple.
>  It's optional, so why not add this
Unless it is a detriment to some other feature or allows cheating in some way, I don't see why shouldn't be added.

Well ... talking generally, we have to also consider whether the feature justifies the extra code and maintenance. Otherwise we would add every 'harmless' optional feature anyone thinks of (we have endless ideas suggested, many silly) and the project becoming unmanageable due to this. Simplicity is a, maybe even 'the', central philosophy of MT, and has to be for a volunteer open source project with very limited core dev time.

**Comment by lhofhansl** [comment]
Tried it. Works well. +1

**Comment by Extex101** [comment]
I'm all for it :+1: as long as only works when pressing foward

**Comment by ClobberXD** [comment]
> Two approvals!

Assuming you were referring to my approval, it doesn't count as an official approval, sorry :P

**Comment by random-geek** [comment]
'Twould be nice if we could get a second approval for this.

**Comment by lhofhansl** [comment]
I'd be happy to merge this... But I can't open travis for some reason, hence I can't see what it is complaining about.

**Comment by paramat** [comment]
Closed and re-opened to re-trigger travis.

**Comment by paramat** [comment]
Not essential for 5.0.0 but should be merged as has 2 approvals.

**Pull request merged**